subsys/bluetooth/host/hci_core.h

```diff
@@ -7,7 +7,10 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/* Skip devicetree.h for unit test as it causes some define collitions. */
+#ifndef ZTEST_UNITTEST
 #include <zephyr/devicetree.h>
+#endif
```

Comment:
Wouldn't it make more sense to have this #ifndef inside the `devicetree.h` file itself, so that not everybody who wants to include it needs to know of this exception?

Reply:
Yeah not liking that either, but hey seems like that override was only needed on the linker file itself, I added a couple of undef for those at the end of the file and that seems to work, so that crappy hack is self contained at least.